src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImage.groovy

```diff
@@ -124,13 +127,13 @@ class DockerBuildImage extends AbstractDockerRemoteApiTask implements RegistryCr
     }
 
     void saveImageId() {
-        TaskStateHelper state = new TaskStateHelper(name, project.buildDir)
+        TaskStateHelper state = new TaskStateHelper(this.class.simpleName, project.buildDir)
         state.put("imageId", getImageId())
     }
 
     boolean previouslyBuiltImageExists() {
         boolean imageExists = false
-        String prevImageId = new TaskStateHelper(name, project.buildDir).get("imageId")
+        String prevImageId = new TaskStateHelper(this.class.simpleName, project.buildDir).get("imageId")
```

Comment:
@bmuschko the imageId is unique which is what I think we really care about. I don't think we care so much about a given user defined task name so much as what its inputs/outputs are. With this new code the user can define any task name they like but it's the inputs, which are created outside of the build by docker if you get what I mean, which need to be unique.